pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -209,6 +220,17 @@ func isErrAndNotRetryAble(statusCode proto.StatusCode) bool {
 		statusCode != proto.StatusCode_STATUSCHECK_DEPLOYMENT_ROLLOUT_PENDING
 }
 
+// AreContainersNotRetryAble goes through all pod statuses and return true
+// if any cannot be retried
+func (d *Deployment) AreContainersNotRetryAble() bool {
```

Comment:
How about `HasEncounteredUnrecoverableError()`?  The containers are the implementation.  Perhaps we'll find that there are other reasons why a deployment should be considered to have failed.